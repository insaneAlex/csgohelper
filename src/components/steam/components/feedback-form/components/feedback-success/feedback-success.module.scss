@import 'styles/variables.scss';

.wrapper {
  width: 500px;
  border: 1px solid $main-font-color;
  border-radius: 5px;
  height: 360px;
  margin: 0 auto;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;

  @include body-short;
}

.title {
  font-size: 20px;
  margin-bottom: 20px;
}

.link {
  background-color: transparent;
  border: 1px solid $darken-light;
  cursor: pointer;
  text-align: center;
  margin: 15px 5px 5px 5px;
  padding: 5px;
  width: 150px;
  &:active {
    transform: scale(0.95);
  }

  @include body-short-link;
}

@media (max-width: 768px) {
  .link {
    width: calc(100% - 40px);
  }
  .wrapper {
    border: none;
  }
}
