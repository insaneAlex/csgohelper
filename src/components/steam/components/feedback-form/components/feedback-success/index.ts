export {FeedbackSuccess} from './feedback-success';
