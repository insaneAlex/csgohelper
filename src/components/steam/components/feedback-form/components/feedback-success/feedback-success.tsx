import {FC} from 'react';
import {useRouter} from 'next/router';
import {Button} from '@/src/components/ui';

import styles from './feedback-success.module.scss';

export const FeedbackSuccess: FC = () => {
  const router = useRouter();

  return (
    <section className={styles.wrapper}>
      <h3 className={styles.title}>Thank you!</h3>
      <p className={styles.text}>Feedback have been successfully sent.</p>

      <Button onClick={() => router.push('/')}>Return to Home</Button>
    </section>
  );
};
