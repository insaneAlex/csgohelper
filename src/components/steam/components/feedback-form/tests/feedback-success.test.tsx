import {fireEvent, render, screen} from '@testing-library/react';
import {FeedbackSuccess} from '../feedback-success';

const routerPushMock = jest.fn();
jest.mock('next/router', () => ({useRouter: () => ({push: routerPushMock})}));

describe('FeedbackSuccess', () => {
  it('should render correctly', () => {
    const {container} = render(<FeedbackSuccess />);
    expect(container).toMatchSnapshot();
  });
  describe('when user clicks "Return to Home" button', () => {
    it('should push "/" route', () => {
      render(<FeedbackSuccess />);
      fireEvent.click(screen.getByRole('button', {name: /Return to Home/i}));
      expect(routerPushMock).toHaveBeenCalledWith('/');
    });
  });
});
