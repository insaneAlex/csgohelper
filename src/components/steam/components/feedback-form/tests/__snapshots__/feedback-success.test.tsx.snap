// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeedbackSuccess should render correctly 1`] = `
<div>
  <section
    class="wrapper"
  >
    <h3
      class="title"
    >
      Thank you!
    </h3>
    <p
      class="text"
    >
      Feedback have been successfully sent.
    </p>
    <button
      class="button"
      type="button"
    >
      Return to Home
    </button>
  </section>
</div>
`;
